File: 0404) Sum of Left Leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        int ans = 0;
        calculate(root, ans);
        return ans;
    }
    void calculate(TreeNode* root, int &ans)
    {
        if(root == nullptr)
        {
            return;
        }
        if(root->left && !root->left->left && !root->left->right)
        {
            ans += root->left->val;
        }
        calculate(root->left, ans);
        calculate(root->right, ans);
    }
};

File: 0404) Sum of Left Leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        int ans = solve(root);
        return ans;
    }
    int solve(TreeNode root)
    {
        int ans = 0;
        if(root == null)
        {
            return 0;
        }
        if(root.left != null && root.left.left == null && root.left.right == null)
        {
            ans += root.left.val;
        }
        ans += solve(root.left);
        ans += solve(root.right);
        return ans;
    }
}
